agent/functional_tests/tests/functionaltests_test.go

```diff
@@ -297,3 +300,127 @@ func TestDockerAuth(t *testing.T) {
+		Config:     &dockerConfig,
+		HostConfig: &dockerHostConfig,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := client.StartContainer(squidContainer.ID, &dockerHostConfig); err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		client.RemoveContainer(docker.RemoveContainerOptions{
+			Force:         true,
+			ID:            squidContainer.ID,
+			RemoveVolumes: true,
+		})
+	}()
+
+	// Resolve the name so we can use it in the link below; the create returns an ID only
+	squidContainer, err = client.InspectContainer(squidContainer.ID)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Squid startup time
+	time.Sleep(1 * time.Second)
+	t.Logf("Started squid container: %v", squidContainer.Name)
+
+	agent := RunAgent(t, &AgentOptions{
+		ExtraEnvironment: map[string]string{
+			"HTTP_PROXY": "squid:3128",
+			"NO_PROXY":   "169.254.169.254,/var/run/docker.sock",
+		},
+		ContainerLinks: []string{squidContainer.Name + ":squid"},
+	})
+	defer agent.Cleanup()
+	agent.RequireVersion(">1.5.0")
+	task, err := agent.StartTask(t, "simple-exit")
+	if err != nil {
+		t.Fatal(err)
+	}
+	// Verify the agent can run a container using the proxy
+	task.WaitStopped(1 * time.Minute)
+
+	// stop the agent, thus forcing it to close its connections; this is neededj
+	// because squid's access logs are written on DC not connect
+	err := agent.StopAgent()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Now verify it actually used the proxy via squids access logs. Get all the
+	// unique addresses that squid proxied for (assume nothing else used the
+	// proxy).
+	// This should be '3' currently, for example I see the following at the time of writing
+	//     ecs.us-west-2.amazonaws.com:443
+	//     ecs-a-1.us-west-2.amazonaws.com:443
+	//     ecs-t-1.us-west-2.amazonaws.com:443
+	// Note, it connects multiple times to the first one which is an
+	// implementation detail we might change/optimize, intentionally dedupe so
+	// we're not tied to that sorta thing
+	// Note, do a docker exec instead of bindmount the logs out because the logs
+	// will not be permissioned correctly in the bindmount. Once we have proper
+	// user namespacing we could revisit this
+	logExec, err := client.CreateExec(docker.CreateExecOptions{
+		AttachStdout: true,
+		AttachStdin:  false,
+		Container:    squidContainer.ID,
+		// Takes a second to flush the file sometimes, so slightly complicated command to wait for it to be written
+		Cmd: []string{"sh", "-c", "FILE=/var/log/squid/access.log; while [ ! -s $FILE ]; do sleep 1; done; cat $FILE"},
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Logf("Execing cat of /var/log/squid/access.log on %v", squidContainer.ID)
+
+	var squidLogs bytes.Buffer
+	err = client.StartExec(logExec.ID, docker.StartExecOptions{
+		OutputStream: &squidLogs,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	for {
+		tmp, _ := client.InspectExec(logExec.ID)
+		if !tmp.Running {
+			break
+		}
+		time.Sleep(100 * time.Millisecond)
```

Comment:
Are you just relying on the overall test timeout to account for something going wrong here?